cdk/src/stacks/budget.ts

```diff
@@ -1,13 +1,84 @@
-import path from 'path';
-import { NestedStack, NestedStackProps } from 'aws-cdk-lib';
-import { CfnInclude } from 'aws-cdk-lib/cloudformation-include';
+import {
+  NestedStack,
+  NestedStackProps,
+  aws_sns as sns,
+  aws_iam as iam,
+  aws_cloudwatch as cw,
+  aws_cloudwatch_actions as actions,
+  aws_budgets as budgets,
+} from 'aws-cdk-lib';
 import { Construct } from 'constructs';
 
 export class BudgetStack extends NestedStack {
   constructor(scope: Construct, id: string, props: NestedStackProps) {
     super(scope, id, props);
-    new CfnInclude(this, 'SuperwerkerTemplate', {
-      templateFile: path.join(__dirname, '..', '..', '..', 'templates', 'budget.yaml'),
+
+    const budgetNotificationTopic = new sns.Topic(this, 'BudgetNotification');
+
+    const budgetNotificationPolicyStatement = new iam.PolicyStatement({
+      actions: ['sns:Publish'],
+      effect: iam.Effect.ALLOW,
+      principals: [new iam.ServicePrincipal('budget.amazonaws.com')],
+      resources: [budgetNotificationTopic.topicArn],
+    });
+
+    const budgetNotificationPolicyDocument = new iam.PolicyDocument({
+      statements: [budgetNotificationPolicyStatement],
+    });
+
+    new sns.TopicPolicy(this, 'BudgetNotificationPolicy', {
+      policyDocument: budgetNotificationPolicyDocument,
+      topics: [budgetNotificationTopic],
+    });
+
+    const budgetAlarm = new cw.Alarm(this, 'Warning', {
+      comparisonOperator: cw.ComparisonOperator.GREATER_THAN_THRESHOLD,
+      alarmDescription: 'Superwerker default budget forecast exceed previous three months',
+      evaluationPeriods: 1,
+      threshold: 0,
+      treatMissingData: cw.TreatMissingData.MISSING,
```

Comment:
maybe let's keep the original comment here also, do we know in the future why use this setting.